ui/src/tool-panel/tool-panel-container.tsx

```diff
@@ -56,6 +56,17 @@ const ToolPanelContainer: React.FC<ToolPanelContainerProps> = ({ setHoveredOverI
         }
     }, [store.linkingState]);
 
+    const onSelectLocationTrackBadge = (locationTrackId: LocationTrackId) => {
```

Comment:
Tämä funkkari päätyi toistaiseksi näinkin korkealle tasolle. Alempi `<ToolPanel>`-komponentti käyttää suoraan `<SwitchInfobox>`-komponenttia (jolle ei ole containeria). Tuo `<SwitchInfobox>` taas käyttää suoraan `<SwitchJointInfobox>`-komponenttia (tälle on containeri, jota ei käytetä' tämän komponentin yhteydessä.

Mietiskeltiin siis jo hieman, että missä delegaatteja pitäisi luoda, niin toistaiseksi tulkinta on containereissa, mutta esimerkiksi tässä komponenttiketjussa tuo containeri+aiempien delegaattien käyttö oli tällä tasolla. 

Ketjuun voisi ainakin nakkaa tuon olemassaolevan `<SwitchJointInfoBoxContainer>`-komponentin, joskin sitä taas käytetään nykyisin vaihdelinkityksen yhteydessä, jossa badgeista ei varmaan haluttaisi edes klikkauksen valitsevan raidetta. Toisaalta tämä on taas käyttäjälle epäintuitiivista, että eri paikassa badgeklikkaus toimiikin...

Tuota container-käytäntöä pitänee pohdiskella vielä himpun lisää.

Reply:
Tämä lienee täällä ihan ok. Näitä valintoja saatettaisiin tehdä joskus muistakin infobokseista, joten tämä on mielestäni ihan hyvä koko toolpanelille läpileikkaavasti tarjoiltuna toiminnallisuutena